Library/Homebrew/extend/os/linux/extend/ENV/super.rb

```diff
@@ -28,4 +41,74 @@ def determine_dynamic_linker_path(formula)
       "#{HOMEBREW_PREFIX}/lib/ld.so"
     end
   end
+
+  # @private
+  def homebrew_extra_pkg_config_paths
+    paths = []
+    if x11?
+      libs = xorg_recursive_deps.map(&:lib)
+      shares = xorg_recursive_deps.map(&:share)
+      libs.each do |l|
+        paths << l/"pkgconfig"
+      end
+      shares.each do |s|
+        paths << s/"pkgconfig"
+      end
+    end
+    paths
+  end
+
+  def homebrew_extra_aclocal_paths
+    paths = []
+    if x11?
+      shares = xorg_recursive_deps.map(&:share)
+      shares.each do |s|
+        paths << s/"aclocal"
+      end
+    end
+    paths
+  end
+
+  def xorg_include_paths
+    xorg_recursive_deps.map(&:include).map(&:to_s)
```

Comment:
As it turns out, converting to strings is unnecessary, and so are the `directory?` checks.

```ruby
class Array
  def to_path_s
    map(&:to_s).uniq.select { |s| File.directory? s }.join(File::PATH_SEPARATOR).chuzzle
  end
end
```

So these are unnecessary for any of these for which `to_path_s` is eventually called.